dataland-e2etests/src/test/kotlin/org/dataland/e2etests/MetaDataControllerTest.kt

```diff
@@ -4,28 +4,46 @@ import org.dataland.datalandbackend.openApiClient.api.CompanyDataControllerApi
 import org.dataland.datalandbackend.openApiClient.api.EuTaxonomyDataControllerApi
 import org.dataland.datalandbackend.openApiClient.api.MetaDataControllerApi
 import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyData
+import org.dataland.datalandbackend.openApiClient.model.CompanyInformation
 import org.dataland.datalandbackend.openApiClient.model.DataMetaInformation
-import org.dataland.datalandbackend.openApiClient.model.PostCompanyRequestBody
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Test
+import java.math.BigDecimal
+import java.time.LocalDate
 
 class MetaDataControllerTest {
 
-    private val basePathToBackendViaProxy = "http://proxy:80/api"
-    private val metaDataControllerApi = MetaDataControllerApi(basePathToBackendViaProxy)
-    private val companyDataControllerApi = CompanyDataControllerApi(basePathToBackendViaProxy)
-    private val euTaxonomyDataControllerApi = EuTaxonomyDataControllerApi(basePathToBackendViaProxy)
+    private val basePathToDatalandProxy = "http://proxy:80/api"
+    private val metaDataControllerApi = MetaDataControllerApi(basePathToDatalandProxy)
+    private val companyDataControllerApi = CompanyDataControllerApi(basePathToDatalandProxy)
+    private val euTaxonomyDataControllerApi = EuTaxonomyDataControllerApi(basePathToDatalandProxy)
+    private val testCompanyList = listOf(
+        CompanyInformation(
+            companyName = "Test-Company_new_1",
+            headquarters = "Test-Headquarters_new_1",
+            industrialSector = "Test-IndustrialSector_new_1",
+            marketCap = BigDecimal(100),
+            reportingDateOfMarketCap = LocalDate.now()
+        ),
+        CompanyInformation(
+            companyName = "Test-Company_new_2",
+            headquarters = "Test-Headquarters_new_2",
+            industrialSector = "Test-IndustrialSector_new_2",
+            marketCap = BigDecimal(200),
+            reportingDateOfMarketCap = LocalDate.now()
+        )
+    )
 
-    private fun populateDatalandWithCompaniesAndEuTaxnomyDataSets(
+    private fun createCompaniesAndEuTaxonomyDataSets(
         numberOfCompanies: Int,
         numberOfDataSetsPerCompany: Int
     ): List<String> {
-        val testCompanyName = "Fantasy-Company_100"
+        val testCompanyInformation = testCompanyList[0]
         val testData = DummyDataCreator().createEuTaxonomyTestDataSet()
 
         val listOfPostedTestCompanyIds = mutableListOf<String>()
         repeat(numberOfCompanies) {
-            val testCompanyId = companyDataControllerApi.postCompany(PostCompanyRequestBody(testCompanyName)).companyId
+            val testCompanyId = companyDataControllerApi.postCompany(testCompanyInformation).companyId
             repeat(numberOfDataSetsPerCompany) {
                 euTaxonomyDataControllerApi.postCompanyAssociatedData(
                     CompanyAssociatedDataEuTaxonomyData(testCompanyId, testData)
```

Comment:
Hier wird eine Company mehrfach hochgeladen und dann auch noch für jede company der gleiche Datensatz mehrfach. Das erscheint mir für Testdaten kein geschickter Ansatz zu sein. Man könnte ja gar nicht unterscheiden ob der richtige Datensatz an der richtigen Stelle gelandet ist. Es steht ja eh überall der gleiche.

Reply:
_Beginne es mir anzuschauen_

Reply:
Ich habe die Funktion so angepasst, dass ein markierender Counter dafür sorgt, dass die Companies und Datensätze sich in den Werten unterscheiden.